wolfcrypt/src/aes.c

```diff
@@ -8970,22 +8970,22 @@ static WARN_UNUSED_RESULT int roll_auth(
     /* encode the length in */
     if (inSz <= 0xFEFF) {
         authLenSz = 2;
-        out[0] ^= ((inSz & 0xFF00) >> 8);
-        out[1] ^=  (inSz & 0x00FF);
+        out[0] ^= (byte)(inSz >> 8);
+        out[1] ^= (byte)inSz;
     }
-    else if (inSz <= 0xFFFFFFFF) {
+    else {
```

Comment:
This changes behavior of the old `else` case?

Reply:
`inSz` is a `word32` so the comparison was a definitional always-true.  it should warn in the future, and the frivolous logic is confusing.  the "why are they doing that?  it doesn't do anything!" kind of confusing.